### PR TITLE
fix: allow for dd/mm/yyyy formats in date picker

### PR DESCRIPTION
**Pull request by wyseguyonline**
This will be ported into compounds upon approval and merge.

### CODE REVIEW

File: demo/demoPages/components/DatePickerPage.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ class DatePickerPage extends Component {
             <h2>DatePicker</h2>
             <DatePicker
               id              = "someGiantId1"
-              dateFormat      = "mm/dd/yyyy"
+              dateFormat      = "dd/mm/yyyy"
```
Comment (StommePoes) [inline demo/demoPages/components/DatePickerPage.js]: I won't complain to setting the default to the version the whole world except the US uses, however I was under the impression that everyone wanted US-oriented defaults.

Comment (wyseguyonline) [reply demo/demoPages/components/DatePickerPage.js]: This is for demo purposes only.

Comment (StommePoes) [reply demo/demoPages/components/DatePickerPage.js]: me gusta

File: src/components/DatePicker/DatePicker.js [hunk 2]
```diff
@@ -161,6 +162,7 @@ function _datePickerOpen() {
 };
 
 function _changeHandler(e) {
+  const { dateFormat } = this.props;);
```
Comment (mhomolak) [inline src/components/DatePicker/DatePicker.js]: Syntax error here when I tried to boot it up.

File: demo/demoPages/components/DatePickerPage.js (hunk 1 above)
Comment (mhomolak) [reply demo/demoPages/components/DatePickerPage.js]: Line 76 has the example string which still reflects the old format for `dateFormat`.

File: src/components/DatePicker/DatePicker.js [hunk 3]
```diff
@@ -171,13 +173,21 @@ function _changeHandler(e) {
...
+
   const changeHandlerParam = {
     target: {
-      value: (date.selectedMonth + 1) + '/' + date.selectedDate + '/' + date.selectedYear
+      value: dateString
     }
   };
+
   this.setState({
-    datepickerValue: moment(date.selectedDt).format('L'),
```
Comment (mhomolak) [inline src/components/DatePicker/DatePicker.js]: Could the dependency on moment be removed now after this change?

**Comment by StommePoes** [comment]
Runs okay over here.

I'm not sure why we're not dogfooding our own dropdown for the dropdown... it stays open when I click the datepicker :/ but this PR looks good from here.

**Comment by umahaea** [comment]
Regressions passed.

**Comment by umahaea** [comment]
merging it.

**Pull request merged**